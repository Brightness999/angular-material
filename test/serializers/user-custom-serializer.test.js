/**
 * @fileOverview Test custom user serializer
 */
const Logality = require('../../app/logality');
const { sink, stubLogality } = require('../lib/tester.lib');

const UDO_MOCK = {
  userId: 10,
  userEmail: 'one@go.com',
};

describe('Custom User Serializer', () => {
  stubLogality();

  test('Will log custom UDO', (done) => {
    const serializers = {
      user: (udo) => {
        return {
          path: 'context.user',
          value: {
            id: udo.userId,
            email: udo.userEmail,
          },
        };
      },
    };

    const logality = new Logality({
      appName: 'testLogality',
      serializers,
      wstream: sink((chunk) => {
        expect(chunk).toMatchSnapshot();
        done();
      }),
    });

    const log = logality.get();

    log('info', 'hello world', { user: UDO_MOCK });
  });

  test('Will log custom UDO on custom path', (done) => {
    const serializers = {
      user: (udo) => {
        return {
          path: 'user',
          value: {
            id: udo.userId,
            email: udo.userEmail,
          },
        };
      },
    };

    const logality = new Logality({
      appName: 'testLogality',
      serializers,
      wstream: sink((chunk) => {
        expect(chunk).toMatchSnapshot();
        done();
      }),
    });

    const log = logality.get();

    log('info', 'hello world', { user: UDO_MOCK });
  });
});
