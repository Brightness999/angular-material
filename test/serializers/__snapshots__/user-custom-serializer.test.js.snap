// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom User Serializer Will log custom UDO 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "testLogality",
    },
    "source": Object {
      "file_name": "/test/serializers/user-custom-serializer.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
    "user": Object {
      "email": "one@go.com",
      "id": 10,
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
  "severity": 6,
}
`;

exports[`Custom User Serializer Will log custom UDO on custom path 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "testLogality",
    },
    "source": Object {
      "file_name": "/test/serializers/user-custom-serializer.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
  "severity": 6,
  "user": Object {
    "email": "one@go.com",
    "id": 10,
  },
}
`;
